Tests/ExtractFilesTests_Swift.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ExtractDataTests.m
+//  UnzipKit
+//
+//  Created by Dov Frankel on 7/16/15.
+//  Copyright (c) 2015 Abbey Code. All rights reserved.
+//
+
+import UnzipKit
+
+class ExtractDataTestsSwift: UZKArchiveTestCase {
+
+    func testExtractData_NoPassword() {
+        let testArchives = ["Test Archive (Password).zip"]
+        for testArchiveName in testArchives {
+            do {
+                let archive = try UZKArchive(URL: self.testFileURLs[testArchiveName])
+                let data = try archive.extractDataFromFile("Test File A.txt", progress: { (percentDecompressed) -> Void in
+                    NSLog("Extracting, \(percentDecompressed) complete");
+                })
+            } catch let error as NSError {
```

Comment:
I don't think that what I suggested is any less "Swifty". This isn't production Swift code, but rather a test to validate output. If you prefer, declare the variable as `var extractError: ErrorType?` instead, but my point was that if the code doesn't correctly write the error, or writes a type other than expected, the tests I suggested will fail. As you wrote them, they would still succeed.

You can try this out - if you update `extractDataFromFile` to return nil on its first line, and run your tests, I believe they will still succeed, as no error will be thrown.

The best way to handle these tests would be to use a better matching framework than `XCTAssert`. Someday I'll switch the tests to use something like [Nimble](https://github.com/Quick/Nimble), which would allow you to write something like:

```Swift
expect{ try archive.extractDataFromFile("Test File A.txt") }.to(throwError {err in
    expect(err.code) == UZKErrorCode.InvalidPassword
})
```

I'd rather not start using Nimble quite yet, though.